File: boj/DFS와BFS/2606_바이러스.cpp
// dfs로 풀이
// cnt로 개수 파악
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;
vector<int> graph[101]; //1부터 저장
bool visited[101];
int cnt=0;
void dfs(int s){
	visited[s]=true;

	for(int i=0;i<graph[s].size();i++){
		int next=graph[s][i];
		if(!visited[next]){
			cnt++;
			dfs(next);
		}
	}
}
int main() {
	int n,m;
	cin>>n>>m;
	
	for(int i=0;i<m;i++){
		int u,v;
		cin>>u>>v;
		
		graph[u].push_back(v);
		graph[v].push_back(u);
	}
	
	for(int i=0;i<n;i++)
		sort(graph[i+1].begin(), graph[i+1].end());
	dfs(1);
	cout<<cnt<<"\n";
	return 0;
}

File: boj/DFS와BFS/2606_바이러스.py
def dfs(graph, v, visited):
	visited[v] = True

	for i in graph[v]:
		if not visited[i]:
			dfs(graph, i, visited)
	
n = int(input())
m = int(input())

graph = [[] for _ in range(n+1)]
visited = [False] * (n+1)

for _ in range(m):
	a, b = map(int, input().rstrip().split())
	graph[a].append(b)
	graph[b].append(a)

for g in graph:
	g.sort()

dfs(graph, 1, visited)

print(visited.count(True)-1) # bool 타입도 count로 셀 수 있다. 방문한 노드 - 1(시작노드 1 빼줌) 출력.